core/drivers/clk/clk_dt.c

```diff
@@ -13,36 +13,36 @@
 #include <libfdt.h>
 #include <stddef.h>
 
-struct clk *clk_dt_get_by_name(const void *fdt, int nodeoffset,
-			       const char *name, TEE_Result *res)
+TEE_Result clk_dt_get_by_name(const void *fdt, int nodeoffset,
+			      const char *name, struct clk **clk)
 {
 	int clk_id = 0;
 
 	clk_id = fdt_stringlist_search(fdt, nodeoffset, "clock-names", name);
 	if (clk_id < 0) {
-		*res = TEE_ERROR_GENERIC;
-		return NULL;
+		*clk = NULL;
+		return TEE_ERROR_GENERIC;
 	}
 
-	return clk_dt_get_by_idx(fdt, nodeoffset, clk_id, res);
+	return clk_dt_get_by_index(fdt, nodeoffset, clk_id, clk);
 }
 
-static struct clk *clk_dt_get_by_idx_prop(const char *prop_name,
-					  const void *fdt, int nodeoffset,
-					  unsigned int clk_idx, TEE_Result *res)
+static TEE_Result clk_dt_get_by_idx_prop(const char *prop_name, const void *fdt,
+					 int nodeoffset, unsigned int clk_idx,
+					 struct clk **clk)
 {
-	void *device = dt_driver_device_from_node_idx_prop(prop_name, fdt,
-							   nodeoffset, clk_idx,
-							   DT_DRIVER_CLK,
-							   res);
+	TEE_Result res = TEE_ERROR_GENERIC;
 
-	return (struct clk *)device;
+	*clk = dt_driver_device_from_node_idx_prop(prop_name, fdt, nodeoffset,
```

Comment:
I suppose the same logic could be applied to `dt_driver_device_from_node_idx_prop()` then, but I guess we can save that for another PR.

Reply:
Yes, `dt_driver_device_from_node_idx_prop()`, could be changed accordingly. Returning `void *` is only needed at `xxx_dt_get()` callback level. I'll address in a next commit.